examples/step-3/doc/results.dox

```diff
@@ -224,3 +224,213 @@ suggestions:
+     @code
+     HDF5 FILE 
+        name /
+    filename 
+
+         name       otype  dclass     dim
+     0 cells       H5I_DATASET INTEGER  x 1024
+     1 mean_value  H5I_DATASET FLOAT   1      
+     2 nodes       H5I_DATASET FLOAT    x 1089
+     3 point_value H5I_DATASET FLOAT   1      
+     4 solution    H5I_DATASET FLOAT    x 1089
+     @endcode
+     The datasets can be accessed by <code>h5f\$name</code>. The function <code>dim(h5f\$cells)</code> gives us
+     the dimensions of our cell matrix. We can see the following three matrices, as well as the two additional
+     data points we added.
+     <ul> 
+     <li> <code>cells</code> a 4x1024 matrix that stores the  (C++) vertex indices for each cell 
+     <li> <code>nodes</code> a 2x1089 matrix storing the position values (x,y) for our cell vertices
+     <li> <code>solution</code> a 1x1089 matrix storing the values of our solution at each vertex
+     </ul>
+     Now we can use this data to generate our various plots. Plotting with ggplot2 usually splits into two steps.
+     At first the data needs to be manipulated and added to a <code>data.frame</code>.
+     After that a <code>ggplot</code> object is constructed and manipulated by adding plot elements to it.
+     <ul>
+     <li>
+     Plotting the grid <br>
+     <code>nodes</code> and <code>cells</code> contain all the information we need to plot our grid. 
+     The following code wraps all the data into one dataframe for plotting our grid
+     @code
+     #counting in R starts at 1 instead of 0 so we need to increment all vertex indices
+     cell_ids <- h5f$cells+1 
+     #store the x and y positions of each vertex in one big vector in a cell by cell fashion 
+     #  (every 4 entries belong to one cell)
+     cells_x <- h5f$nodes[1,][cell_ids]
+     cells_y <- h5f$nodes[2,][cell_ids]
+     #construct a vector that stores the matching cell by cell grouping (1,1,1,1,2,2,2,2,...)
+     groups <- rep(1:ncol(cell_ids),each=4)
+     #finally put everything into one dataframe   
+     meshdata <- data.frame(x = cells_x, y = cells_y, id = groups)
+     @endcode
+     With the finished dataframe we have everything we need to plot our grid
+     @code
+     pdf (paste("grid_",refinement,".pdf",sep=""),width = 5,height = 5) #Open new PDF file
+     plt <- ggplot(meshdata,aes(x=x,y=y,group=id)) #construction of our plot object, at first only data 
+     plt <- plt + geom_polygon(fill="white",colour="black") #actual plotting of the grid as polygons
+     plt <- plt + ggtitle(paste("grid at refinement level #",refinement))
+     print(plt) #show the current state of the plot/ add it to the pdf
+     dev.off() #close PDF file
+     @endcode
+     <table width="60%" align="center">
+       <tr>
+         <td align="center">
+           <img src="https://www.dealii.org/images/steps/developer/step-3.extensions.grid_5.png" alt="Grid after 5 refinement steps of step-3">
+         </td>
+       </tr>
+     </table>
+     <li> 
```

Comment:
For example, deleting this and the next line and starting the paragraph with something like "To make a 2D pseudocolor plot of our solution"